docs/data-cookbooks/cnm-workflow.md

```diff
@@ -0,0 +1,166 @@
+
+This can be done via the [Cumulus Dashboard](https://github.com/nasa/cumulus-dashboard) if installed or the [API](../api.md).  It is strongly recommended to use the dashboard if possible.
+
+### Workflow Configurations
+
+For our initial example, we're going to trigger HelloWorld workflow that is provided in the example deployment.
+
+The [workflow definition](../workflows/README.md) can be found in [cumulus/example/workflows.yml](https://github.com/nasa/cumulus/blob/master/example/workflows.yml)  under `HelloWorldWorkflow:`
+
+(Please note this snippet is included as a sample only, please review the source file for the most up-to-date workflow)
+
+```
+HelloWorldWorkflow:
+  Comment: 'Returns Hello World'
+  StartAt: StartStatus
+  States:
+    StartStatus:
+      Type: Task
+      Resource: ${SfSnsReportLambdaFunction.Arn}
+      CumulusConfig:
+        cumulus_message:
+          input: '{$}'
+      Next: HelloWorld
+    HelloWorld:
+      CumulusConfig:
+        buckets: '{$.meta.buckets}'
+        provider: '{$.meta.provider}'
+        collection: '{$.meta.collection}'
+      Type: Task
+      Resource: ${HelloWorldLambdaFunction.Arn}
+      Next: StopStatus
+    StopStatus:
+      Type: Task
+      Resource: ${SfSnsReportLambdaFunction.Arn}
+      CumulusConfig:
+        sfnEnd: true
+        stack: '{$.meta.stack}'
+        bucket: '{$.meta.buckets.internal.name}'
+        stateMachine: '{$.cumulus_meta.state_machine}'
+        executionName: '{$.cumulus_meta.execution_name}'
+        cumulus_message:
+          input: '{$}'
+      Catch:
+        - ErrorEquals:
+          - States.ALL
+          Next: WorkflowFailed
+      End: true
+    WorkflowFailed:
+      Type: Fail
+      Cause: 'Workflow failed'
+```
+If you're using the example deployment this workflow will already be included.  If you're attempting to use this example in a custom deployment you'll need to include the HelloWorld in your workflows.yml, then re-deploy to utilize it.
+
+### Task Configuration
+
+This cookbook entry assumes the HelloWorld [task](../workflows/developing-workflow-tasks.md) is defined in the `lambdas.yml` configuration file under `HelloWorld:`
+
+```
+HelloWorld:
+  handler: index.handler
+  timeout: 300
+  memory: 256
+  source: 'node_modules/@cumulus/hello-world/dist/'
+  useMessageAdapter: true
+```
+
+If you are using the example deployment this task will already be included in the configuration file.
+
+If you are attempting to use this example in a custom deployment you'll need to include the HelloWorld task in your lambdas.yml, then re-deploy to utilize it.
+
+##### Rule Configuration
+
+Cumulus provides a built-in Kinesis consumer lambda function ([kinesis-consumer](https://github.com/nasa/cumulus/blob/master/packages/api/lambdas/kinesis-consumer.js)) that will read events off of a preconfigured Kinesis stream and trigger a workflow when a Cumulus rule is configured via the Cumulus dashboard or API.
```

Comment:
This to me seems a little too in the weeds to link to the actual code. I think the users just need to know that something will pick up what they put on the kinesis stream.

Reply:
I do have a bit of uncertainty surrounding the target audience's interests.   With my "technical DAAC user" hat on, being able to peruse the lambda source probably would have been helpful several times in the past for context, but I agree this probably is a bit over the noise threshold  for some users.  We can remove it.

Reply:
I know that as someone who has worked on a project that did some integrating with Cumulus, having access to the code and going through source was undeniably valuable. I think having links to actual code somewhere in the documentation would be really useful. I can't speak to whether or not that belongs in the workbooks, but it would be much simpler for the more technical users than expecting them to dig through our repo.

Reply:
Ok I get that and appreciate the perspective. I guess the question is, is that something for the data cookbook? @bilts